src/emoji/data.js

```diff
@@ -28,12 +28,34 @@ export const getFilteredEmojis = (
   query: string,
   activeImageEmojiByName: $ReadOnly<{ [string]: ImageEmojiType }>,
 ): $ReadOnlyArray<{| emoji_type: EmojiType, name: string, code: string |}> => {
+  const matchingUnicodeEmoji = Object.entries(unicodeCodeByName).map(x => {
+    if (typeof x[1] !== 'string') {
+      // This is impossible, but flow isn't smart enough to figure that out.
+      throw new Error('Emoji name is not a string');
+    }
+    // We may want to have fancier emoji-matching logic in the future: this
+    // code will not work with skin-tone modifiers, for instance, and
+    // there may be other strange unicode things that warrant special
+    // handling. It would be simple to add code to strip off the
+    // skin-tone modifiers in the query, but it seems worse to suggest to
+    // the user the default skin color emoji when they've typed in a
+    // modified one than to display nothing. We should circle back to
+    // this later, and figure out what our overall strategy for skin-tone
+    // modifiers across the app should be.
+    const matches_emoji_literal = parseUnicodeEmojiCode(x[1]) === query;
+    const matches_emoji_name = Math.min(1, x[0].indexOf(query));
+    return [x[0], matches_emoji_literal ? 0 : matches_emoji_name];
+  });
+
+  const matchingCustomEmoji = Object.keys(activeImageEmojiByName).map(x => [
```

Comment:
Let's call this "image" emoji rather than "custom". That matches the name of the data it's processing, `activeImageEmojiByName`... and it also reflects the fact that it includes not only the custom emoji but also the Zulip extra emoji. (Of which there's just one: `:zulip:`.)